Purchases/SwiftSources/Purchasing/TransactionsFactory.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  PurchaserInfoHelper.swift
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCTransactionsFactory) public class TransactionsFactory: NSObject {
+
+    @objc public func nonSubscriptionTransactions(with subscriptionsData: [String: [[String: Any]]], dateFormatter: DateFormatter) -> [Transaction] {
+        subscriptionsData.flatMap { (productId: String, transactionData: [[String: Any]]) -> [Transaction] in
```

Comment:
I usually don't specify the types in these since they can be inferred, but In this case I like it - it helps the compiler and the dev not get lost in the nested block.